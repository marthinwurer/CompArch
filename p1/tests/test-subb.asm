; test the subb instruction
	.entry	start
;
	.org	0x50
one:	.word	0x00001
five:	.word	0x00005
neg1:	.word	0xfffff
neg5:	.word	0xffffb
;
	.org	0x100
start:	ldb	one
	subb	one
;
	ldb	one
	subb	five
	subb	one
;
	ldb	one
	subb	neg1
;
	ldb	one
	subb	neg5
;
	halt
