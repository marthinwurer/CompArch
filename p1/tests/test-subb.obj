50 1 00001
51 1 00005
52 1 fffff
53 1 ffffb
100 1 11050
101 1 23050
102 1 11050
103 1 23051
104 1 23050
105 1 11050
106 1 23052
107 1 11050
108 1 23053
109 1 00000
100
