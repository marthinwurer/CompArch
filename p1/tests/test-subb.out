CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 100
100:  11050  LDB   050  A[00000]  B[00000]  XR[000]
101:  23050  SUBB  050  A[00000]  B[00001]  XR[000]
102:  11050  LDB   050  A[00000]  B[00000]  XR[000]
103:  23051  SUBB  051  A[00000]  B[00001]  XR[000]
104:  23050  SUBB  050  A[00000]  B[ffffc]  XR[000]
105:  11050  LDB   050  A[00000]  B[ffffb]  XR[000]
106:  23052  SUBB  052  A[00000]  B[00001]  XR[000]
107:  11050  LDB   050  A[00000]  B[00002]  XR[000]
108:  23053  SUBB  053  A[00000]  B[00001]  XR[000]
109:  00000  HALT       A[00000]  B[00006]  XR[000]

Machine Halted - HALT instruction executed

Simulated time 53 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
